toffy/streak_detection_test.py

```diff
@@ -0,0 +1,445 @@
+        "streak_mask": Read_Fn(io.imread, ".tiff"),
+        "streak_df": Read_Fn(pd.read_csv, ".csv"),
+        "filtered_streak_mask": Read_Fn(io.imread, ".tiff"),
+        "filtered_streak_df": Read_Fn(pd.read_csv, ".csv"),
+        "boxed_streaks": Read_Fn(io.imread, ".tiff"),
+        "corrected_streak_mask": Read_Fn(io.imread, ".tiff"),
+    }
+
+    #  Test `streak_detection::_save`
+    for field, rf in fields.items():
+        sd._save(streak_data=streak_data_test, name=field)
+
+        if rf.ext == ".csv":
+            data = rf.read_fn(Path(tmp_path, f"streak_data_chan0", field + rf.ext), index_col=0)
+        else:
+            data = rf.read_fn(Path(tmp_path, f"streak_data_chan0", field + rf.ext))
+
+        if type(data) is np.ndarray:
+            assert np.array_equal(data, getattr(streak_data_test, field))
+        if type(data) is pd.DataFrame:
+            pd.testing.assert_frame_equal(data, getattr(streak_data_test, field))
+
+
+def test_save_streak_masks(tmp_path, streak_dataset):
+    # * Minimum Data Needed:
+    #   1. streak_data with the fields: corrected_dir, streak_channel, and data (streak_mask,
+    #   streak_df, filtered_streak_mask, filtered_streak_df, boxed_streaks, corrected_streak_mask)
+
+    streak_data_test = streak_dataset(
+        corrected_dir=tmp_path, fov="fov0", chan="chan0", shape=(1000, 1000)
+    )
+
+    sd._save_streak_masks(streak_data=streak_data_test)
+
+    Read_Fn = namedtuple("Read_Operation", "read_fn, ext")
+    fields = {
+        "streak_mask": Read_Fn(io.imread, ".tiff"),
+        "streak_df": Read_Fn(pd.read_csv, ".csv"),
+        "filtered_streak_mask": Read_Fn(io.imread, ".tiff"),
+        "filtered_streak_df": Read_Fn(pd.read_csv, ".csv"),
+        "boxed_streaks": Read_Fn(io.imread, ".tiff"),
+        "corrected_streak_mask": Read_Fn(io.imread, ".tiff"),
+    }
+
+    #  Test `streak_detection::_save`
+    for field, rf in fields.items():
+        if rf.ext == ".csv":
+            data = rf.read_fn(Path(tmp_path, f"streak_data_chan0", field + rf.ext), index_col=0)
+        else:
+            data = rf.read_fn(Path(tmp_path, f"streak_data_chan0", field + rf.ext))
+
+        if type(data) is np.ndarray:
+            assert np.array_equal(data, getattr(streak_data_test, field))
+        if type(data) is pd.DataFrame:
+            pd.testing.assert_frame_equal(data, getattr(streak_data_test, field))
+
+
+def test_make_binary_mask():
+    # * Minimum Data Needed
+    #   1. Numpy array
+    # Only need to test to make sure the dimensions are the same, and the output is a binary array
+    rng = np.random.default_rng(12345)
+    row_size = 1000
+    col_size = 1000
+    input_image_test = rng.integers(low=0, high=16, size=(row_size, col_size))
+
+    binary_mask = sd._make_binary_mask(input_image=input_image_test)
+    assert np.all(np.isin(binary_mask, [0, 1]))
+
+
+def test_make_mask_dataframe():
+    # * Minimum Data Needed
+    #   1. streak_data: Only need `streak_mask`
```

Comment:
Can remove these comments describing minimum data in favor of describing what is being tested and why. The rest of this function has good level of commenting, some of the earlier ones are a bit more sparse